packages/salesforcedx-vscode-core/src/commands/forceSandboxCreate.ts

```diff
@@ -0,0 +1,212 @@
+  SfdxWorkspaceChecker
+} from './util';
+
+export const DEFAULT_ALIAS = 'vscodeSandbox';
+export const DEFAULT_WAIT_TIME_MINS = '30';
+
+export class ForceSandboxCreateExecutor extends SfdxCommandletExecutor<
+  AliasAndFileSelection
+  > {
+  public build(data: AliasAndFileSelection): Command {
+    const selectionPath = path.relative(
+      getRootWorkspacePath(), // this is safe because of workspaceChecker
+      data.file
+    );
+    return new SfdxCommandBuilder()
+      .withDescription(
+        nls.localize('force_sandbox_create')
+      )
+      .withArg('force:org:create')
+      .withFlag('--type', 'sandbox')
+      .withFlag('--definitionfile', `${selectionPath}`)
+      .withFlag('-a', data.alias)
+      .withFlag('-w', data.waitTime)
+      .withLogName('force_create_sandbox_org')
+      .withJson()
+      .build();
+  }
+
+  public execute(response: ContinueResponse<AliasAndFileSelection>): void {
+    const startTime = process.hrtime();
+    const cancellationTokenSource = new vscode.CancellationTokenSource();
+    const cancellationToken = cancellationTokenSource.token;
+    const execution = new CliCommandExecutor(this.build(response.data), {
+      cwd: getRootWorkspacePath(),
+      env: { SFDX_JSON_TO_STDOUT: 'true' }
+    }).execute(cancellationToken);
+
+    channelService.streamCommandStartStop(execution);
+
+    let stdOut = '';
+    execution.stdoutSubject.subscribe(realData => {
+      stdOut += realData.toString();
+    });
+
+    execution.processExitSubject.subscribe(async exitCode => {
+      this.logMetric(execution.command.logName, startTime);
+      try {
+        const createParser = new OrgCreateResultParser(stdOut);
+
+        if (createParser.createIsSuccessful()) {
+          // NOTE: there is a beta in which this command also allows users to create sandboxes
+          // once it's GA this will have to be updated
+          setWorkspaceOrgTypeWithOrgType(OrgType.SourceTracked);
+        } else {
+          const errorResponse = createParser.getResult() as OrgCreateErrorResult;
+          if (errorResponse) {
+            channelService.appendLine(errorResponse.message);
+            telemetryService.sendException(
+              'force_org_create',
+              errorResponse.message
+            );
+          }
+        }
+      } catch (err) {
+        channelService.appendLine(
+          nls.localize('force_org_create_result_parsing_error')
+        );
+        channelService.appendLine(err);
+        telemetryService.sendException(
+          'force_org_create',
+          `Error while parsing org create response ${err}`
+        );
+      }
+    });
+
+    notificationService.reportCommandExecutionStatus(
+      execution,
+      cancellationToken
+    );
+    ProgressNotification.show(execution, cancellationTokenSource);
+    taskViewService.addCommandExecution(execution, cancellationTokenSource);
+  }
+}
+
+export class SandboxCreationGatherer implements ParametersGatherer<Alias> {
```

Comment:
This, the `SandboxCloneGatherer` on the other command in this PR and `AliasGatherer` in the normal org create use the same alias logic. Is there value in having the alias part be something that gets extended and used the same way in these commands ? 

https://github.com/forcedotcom/salesforcedx-vscode/blob/94870434ad524162cc35b5e85313e8b0c36e6ef8/packages/salesforcedx-vscode-core/src/commands/forceOrgCreate.ts#L129

Reply:
I can look into it. The parameters are quite different though.